### PR TITLE
feat: factor vote-aggregation into it's own pkg [BLO-814]

### PR DESCRIPTION
**Pull request by g3ndo9kar9**
## In This PR
- I factor the logic for aggregating votes from an extended commit to the `strategies/aggregator` pkg
    - I introduce a `VoteAggregator` abstraction to handle this -> we can export this into the dydx PrepareProposal
- I also move all constants used in ABCI into a `slinky/abci/types` pkg

### CODE REVIEW

File: abci/strategies/aggregator/vote_aggregator.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+			dva.logger.Debug(
+				"failed to decode price",
+				"currency_pair_id", cpID,
+				"err", err,
+			)
+
+			// If the price cannot be decoded, continue.
+			//
+			// TODO: Should we return an error here instead?
+			continue
```
Comment (aljo242) [inline abci/strategies/aggregator/vote_aggregator.go:194]: is this a carryover or a new TODO?

**Review by aljo242 — COMMENTED** [review]
super clean !

File: abci/strategies/aggregator/vote_aggregator.go (hunk 1 above)
Comment (g3ndo9kar9) [reply abci/strategies/aggregator/vote_aggregator.go:194]: ya this is a carry-over

Comment (g3ndo9kar9) [reply abci/strategies/aggregator/vote_aggregator.go:194]: Would you prefer that we remove the TODO or change the behavior?

Comment (g3ndo9kar9) [reply abci/strategies/aggregator/vote_aggregator.go:194]: Linear: https://linear.app/skip/issue/BLO-865/resolve-potential-errors-in-preblock-when-aggregating-votes-prices

**Review by aljo242 — APPROVED** [review]

**Review by davidterpay — APPROVED** [review]

**Pull request merged**